client/src/components/Signup/Signup.js

```diff
@@ -0,0 +1,98 @@
+import * as Yup from "yup";
+import { Formik, Form, Field, ErrorMessage } from "formik";
+import { AiOutlineArrowRight } from "react-icons/ai";
+import "./Signup.css";
+const SignUp = () => {
+  const initialValues = {
+    name: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+  };
+
+  const validationSchema = Yup.object().shape({
+    name: Yup.string()
+      .min(3, "Name must be at least 3 characters")
+      .max(30, "Name must be at most 30 characters")
+      .required("Please enter your name"),
+    email: Yup.string()
+      .email("Invalid email address")
+      .required("Please enter your email"),
+    password: Yup.string()
+      .min(10)
+      .required("Please enter your password")
+      .matches(
+        /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$/,
+        "Must contain at least 10 character and a special character"
+      ),
+
+    confirmPassword: Yup.string()
+      .oneOf([Yup.ref("password"), null], "Passwords must match")
+      .required("Please confirm the password"),
+  });
+
+  const onSubmit = (values, { setSubmitting }) => {
+    setTimeout(() => {
+      console.log(values);
+      sessionStorage.setItem("name", values.name);
+
+      setSubmitting(false);
+      window.open("http://localhost:3000/", "_self");
```

Comment:
`http://localhost:3000/` -> process.env.Domain

Reply:
Sure, I will do it in next PR